### ISSUE TITLE
Fatal Exception: NSInvalidArgumentException or EXC_BAD_ACCESS in [MQTTDecoder close]

### ISSUE THREAD

**Issue by castus** [aws/aws-sdk-ios#904]
I have a great problem here. Actually first time in my life I have stability alert from Crashlytics. App starts to crash like crazy. From the stacktrace I see that when I invoke disconnect the app will crash. This bug is very hard to reproduce, in fact I cannot invoke it, but I see errors in Crashlytics.

Stacktrace:
```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x18509f164 __exceptionPreprocess
1  libobjc.A.dylib                0x1842e8528 objc_exception_throw
2  CoreFoundation                 0x1850ac628 __methodDescriptionForSelector
3  CoreFoundation                 0x1850a4b10 ___forwarding___
4  CoreFoundation                 0x184f89ccc _CF_forwarding_prep_0
5  AWSIoT                         0x1010d9110 -[MQTTDecoder close] (MQTTDecoder.m:58)
6  AWSIoT                         0x1010dc1ac -[MQTTSession close] (MQTTSession.m:224)
7  AWSIoT                         0x1010c49e4 -[AWSIoTMQTTClient disconnect] (AWSIoTMQTTClient.m:415)
8  AWSIoT                         0x1010abd40 -[AWSIoTDataManager disconnect] (AWSIoTDataManager.m:370)
9  MyApp                    0x100b6975c closure #1 in LiveSynchronizationController.disconnect() (LiveSynchronizationController.swift:80)
10 MyApp                    0x100a46e5c _T0Ieg_IeyB_TR (DashboardViewController.swift)
11 libdispatch.dylib              0x184a1ea54 _dispatch_call_block_and_release
12 libdispatch.dylib              0x184a1ea14 _dispatch_client_callout
13 libdispatch.dylib              0x184a25bc8 _dispatch_queue_override_invoke$VARIANT$mp
14 libdispatch.dylib              0x184a2bcf4 _dispatch_root_queue_drain
15 libdispatch.dylib              0x184a2ba38 _dispatch_worker_thread3
16 libsystem_pthread.dylib        0x184cc706c _pthread_wqthread
17 libsystem_pthread.dylib        0x184cc6b6c start_wqthread
```

Exception:
```
Fatal Exception: NSInvalidArgumentException
-[NSTaggedPointerString removeFromRunLoop:forMode:]: unrecognized selector sent to instance 0xa000000010275f84
```

* What service are you using?
AWSIoT

* In what version of SDK are you facing the problem?
2.6.11

* Is the issue limited to Simulators / Actual Devices?
I see this on Actual Devices

* Can your problem be resolved if you bump to a higher version of SDK?
I've bumped it, but my worries is that it won't fix it.

* Is this problem related to specific iOS version?
No

* How are you consuming the SDK? CocoaPods / Carthage / Prebuilt frameworks?
CocoaPods

* Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.
I don't have a clue. I see a lots of errors from Crashlytics.

Please help 🙏

**Comment by kvasukib** [comment]
Thank you @castus for reporting to us. Sorry for the inconvenience caused. Can you share us the code snippet if you have one? Did you upgrade to 2.6.11 recently and started noticing the crash? Have you encountered this issues in the prior releases?

**Comment by castus** [comment]
Hi @kvasukib ,

The singleton that I use for connection is here https://gist.github.com/castus/fea6d241f967d8cc63aeb009142f1c65
And the usage is here https://gist.github.com/castus/03ec6d8bf5e0f4167e6d4dfe9437db37

This is my first release of AWSIoT so I don't have the comparison.

**Comment by steelzeh** [comment]
I'm also experiencing this.

**Comment by cbommas** [comment]
@castus , @steelzeh

We made a number of crash fixes in 2.6.19. Can you upgrade to the latest and let us know how you fare?
